amethyst_renderer/src/pass/quad/interleaved.rs

```diff
@@ -169,18 +190,74 @@ impl Pass for DrawSprite {
 }
 
 #[derive(Clone, Debug)]
-struct SpriteDrawData {
-    texture: Handle<Texture>,
-    render: SpriteRender,
-    transform: GlobalTransform,
+enum QuadDrawData {
+    Sprite {
+        texture_handle: Handle<Texture>,
+        render: SpriteRender,
+        transform: GlobalTransform,
+    },
+    Image {
+        texture_handle: Handle<Texture>,
+        render: ImageRender,
+        transform: GlobalTransform,
+        width: usize,
+        height: usize,
+    },
```

Comment:
Notice how the fields are duplicated. That's what the unified 2d render rfc fixes.

For this pr, we can keep it like this without an issue.